### ISSUE TITLE
Need to fail-fast on deprecations

### ISSUE THREAD

**Issue by EricCousineau-TRI** [RobotLocomotion/drake-external-examples#159]
Deprecations generally slip through, and cause breakage when their removed.

**Comment by jwnimmer-tri** [comment]
I broke drake-external-examples master again because of this via https://reviewable.io/reviews/RobotLocomotion/drake/12680. Drake removed a deprecated text_logging_gflags header, but it remained used by drake-external-examples until it was fully removed and thus broke the build.

**Comment by EricCousineau-TRI** [comment]
Per f2f: C++ should be easy (`-Wdeprecations` or what not), if we're not already doing that.

For Python, will be annoying (env vars with test runners, updating each testing mode, or parsing stdout), but doable.

**Comment by jwnimmer-tri** [comment]
I think when I work on https://github.com/RobotLocomotion/drake/issues/19167 soon, I should circle back and toggle it here as well.

**Comment by tom-osika** [comment]
With https://github.com/RobotLocomotion/drake/pull/24406 merged - it seems like this essentially boils down to adding `"-Wdeprecated-declarations"` for the C++, and `DRAKE_DEPRECATION_RUNTIME_SEVERITY="error"` for the python examples.

**Comment by jwnimmer-tri** [comment]
Yes.  For the Bazel examples, both settings should go in the `.bazelrc` -- the former as `build --copt=-Werror=deprecated-declarations`, the latter as `build --test_env=DRAKE_DEPRECATION_RUNTIME_SEVERITY=error`.  We'll want a comment nearby along the lines of "We set these in our examples to ensure they stay up-to-date with latest Drake.  You might decide to remove them or set to 'ignore' in your own project."
